x-pack/plugins/apm/server/routes/alerts/rule_types/utils/get_parsed_filtered_query.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+
+export const getParsedFilterQuery: (
+  filter: string | undefined
+) => Array<Record<string, any>> = (filter) => {
+  if (!filter) return [];
+
+  try {
+    const parsedQuery = toElasticsearchQuery(fromKueryExpression(filter));
+    return [parsedQuery];
+  } catch (error) {
+    return [];
+  }
+};
```

Comment:
Can we use [`kqlQuery`](https://github.com/elastic/kibana/blob/933d97ff6d6c90976e947e088708df4127954a87/x-pack/plugins/observability/server/utils/queries.ts#L63) instead?

```suggestion

```

Reply:
Refactored the code to reuse `getParsedFilterQuery` from `observability` plugin.